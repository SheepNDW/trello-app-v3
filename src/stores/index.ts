import { ref } from 'vue'
import { defineStore } from 'pinia'
import type { CurrentEditTask, TaskItem, TasksList } from '@/types'
import { randomUUID } from '@/utils'

export const defaultList: TasksList = [
  {
    id: '61bblwnux0tlb1vp3qk',
    title: 'Hello Vue.js!',
    tasks: [
      {
        id: '79bcdufvlfdlb1voaq5',
        title: 'Vue.js 測試',
        content: '內文區塊',
      },
      {
        id: '4ffywwh2sfelb1vob44',
        title: 'Vue.js 測試2',
        content: '內文區塊2',
      },
    ],
  },
  {
    id: '1ix2wxwsldblb1vpa4h',
    title: 'Hello Laravel!',
    tasks: [
      {
        id: 'zm3frri5a4lb0l8c2e',
        title: 'Laravel 測試',
        content: '內文區塊',
      },
      {
        id: 'b8ijxyk2o9lb1vnzio',
        title: 'Laravel 測試2',
        content: '內文區塊2',
      },
    ],
  },
]

export const useStore = defineStore('store', () => {
  const lists = ref<TasksList>(defaultList)

  const currentEditTask = ref<CurrentEditTask | null>(null)

  const updateListTitle = (cardId = '', title = '') => {
    const card = lists.value.find(list => list.id === cardId)
    if (card) card.title = title
  }

  const addTask = (cardId = '', title = '') => {
    if (!cardId || !title) return

    const card = lists.value.find(list => list.id === cardId)
    card?.tasks.push({
      id: randomUUID(),
      title,
      content: '',
    })
  }

  const openEditTask = (cardId: string, taskId: string) => {
    const card = lists.value.find(list => list.id === cardId)!
    const task = card.tasks.find(task => task.id === taskId)!

    currentEditTask.value = {
      cardId,
      ...task,
    }
  }

  const closeEditTask = () => {
    currentEditTask.value = null
  }

  const updateTask = (cardId: string, taskId: string, title = '', content = '') => {
    const card = lists.value.find(list => list.id === cardId)
    const task = card?.tasks.find(task => task.id === taskId) as TaskItem

    task.title = title
    task.content = content
    closeEditTask()
  }

  return {
    lists,
    updateListTitle,
    addTask,
    currentEditTask,
    openEditTask,
    closeEditTask,
    updateTask,
  }
})
