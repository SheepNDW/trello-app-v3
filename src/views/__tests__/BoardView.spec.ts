import { describe, expect, it, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'
import BoardViewVue from '../BoardView.vue'
import { defaultList } from '@/stores'
import router from '@/router'

describe('BoardView.vue', () => {
  it('renders correctly', () => {
    const wrapper = mount(BoardViewVue, {
      global: {
        plugins: [
          createTestingPinia({
            initialState: {
              lists: defaultList,
            },
            createSpy: vi.fn,
          }),
          router,
        ],
        stubs: { RouterView: true },
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
