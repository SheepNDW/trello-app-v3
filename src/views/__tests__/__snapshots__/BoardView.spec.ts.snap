// Vitest Snapshot v1

exports[`BoardView.vue > renders correctly 1`] = `
"<div class=\\"bg-emerald-700 h-[100vh] w-full block overflow-x-auto overflow-y-hidden\\">
  <div id=\\"board-wrapper\\" class=\\"h-full w-full p-4 block overflow-auto\\">
    <div class=\\"flex flex-row items-start\\">
      <div data-test=\\"root\\" class=\\"bg-slate-200 block border rounded-sm p-2 mx-2 border-gray-500 min-w-[300px]\\" data-draggable=\\"true\\">
        <!-- column -->
        <div class=\\"block overflow-hidden text-ellipsis w-4/5 text-lg\\" data-test=\\"titleText\\">Hello Vue.js!</div><!-- tasks -->
        <div>
          <div class=\\"block my-2 w-full py-2 px-3 bg-white overflow-hidden select-none\\" data-draggable=\\"true\\">
            <div class=\\"block font-bold\\">Vue.js 測試</div>
            <div class=\\"overflow-hidden text-ellipsis\\">內文區塊</div>
          </div>
          <div class=\\"block my-2 w-full py-2 px-3 bg-white overflow-hidden select-none\\" data-draggable=\\"true\\">
            <div class=\\"block font-bold\\">Vue.js 測試2</div>
            <div class=\\"overflow-hidden text-ellipsis\\">內文區塊2</div>
          </div>
        </div><!-- tasks -->
        <!-- add new task -->
        <div class=\\"my-3\\" data-test=\\"root\\">
          <div data-test=\\"addTaskBtn\\" class=\\"bg-slate-200 p-2 hover:bg-slate-300 cursor-pointer text-slate-500\\"> + 點擊以新增任務 </div>
        </div><!-- add new task -->
      </div>
      <div data-test=\\"root\\" class=\\"bg-slate-200 block border rounded-sm p-2 mx-2 border-gray-500 min-w-[300px]\\" data-draggable=\\"true\\">
        <!-- column -->
        <div class=\\"block overflow-hidden text-ellipsis w-4/5 text-lg\\" data-test=\\"titleText\\">Hello Laravel!</div><!-- tasks -->
        <div>
          <div class=\\"block my-2 w-full py-2 px-3 bg-white overflow-hidden select-none\\" data-draggable=\\"true\\">
            <div class=\\"block font-bold\\">Laravel 測試</div>
            <div class=\\"overflow-hidden text-ellipsis\\">內文區塊</div>
          </div>
          <div class=\\"block my-2 w-full py-2 px-3 bg-white overflow-hidden select-none\\" data-draggable=\\"true\\">
            <div class=\\"block font-bold\\">Laravel 測試2</div>
            <div class=\\"overflow-hidden text-ellipsis\\">內文區塊2</div>
          </div>
        </div><!-- tasks -->
        <!-- add new task -->
        <div class=\\"my-3\\" data-test=\\"root\\">
          <div data-test=\\"addTaskBtn\\" class=\\"bg-slate-200 p-2 hover:bg-slate-300 cursor-pointer text-slate-500\\"> + 點擊以新增任務 </div>
        </div><!-- add new task -->
      </div>
      <div class=\\"bg-slate-200 block border rounded-sm p-2 mx-2 border-gray-500 w-[300px] cursor-pointer bg-opacity-70 hover:bg-opacity-90 min-w-[300px]\\">
        <div data-test=\\"addCardBtn\\" class=\\"block select-none\\"> + 新增其他列表 </div>
      </div>
    </div>
  </div>
</div>
<!-- lightbox -->
<router-view-stub name=\\"default\\"></router-view-stub>"
`;
